lib/services/platform-service.ts

```diff
@@ -530,19 +532,19 @@ export class PlatformService implements IPlatformService {
 		return null;
 	}
 
-	public async cleanDestinationApp(platform: string): Promise<void> {
-		await this.ensurePlatformInstalled(platform);
+	public async cleanDestinationApp(platform: string, appFilesUpdaterOptions: IAppFilesUpdaterOptions, platformTemplate: string, projectData: IProjectData): Promise<void> {
+		await this.ensurePlatformInstalled(platform, platformTemplate, projectData);
 
-		const appSourceDirectoryPath = path.join(this.$projectData.projectDir, constants.APP_FOLDER_NAME);
-		let platformData = this.$platformsData.getPlatformData(platform);
+		const appSourceDirectoryPath = path.join(projectData.projectDir, constants.APP_FOLDER_NAME);
+		let platformData = this.$platformsData.getPlatformData(platform, projectData);
 		let appDestinationDirectoryPath = path.join(platformData.appDestinationDirectoryPath, constants.APP_FOLDER_NAME);
-		const appUpdater = new AppFilesUpdater(appSourceDirectoryPath, appDestinationDirectoryPath, this.$options, this.$fs);
+		const appUpdater = new AppFilesUpdater(appSourceDirectoryPath, appDestinationDirectoryPath, appFilesUpdaterOptions, this.$fs);
```

Comment:
I am a bit reluctant to change this in this PR as `AppFilesUpdater` class has a `options` dependency which isn't `$options` and more importantly should be resolved as `$options`.
If it is okay with you I'm willing to make a subsequent PR @rosen-vladimirov